src/main/java/com/mg105/entities/GameState.java

```diff
@@ -139,6 +139,15 @@ public void removeCurrEncounter(){
         this.currEncounter = null;
     }
 
+    /**
+     * Returns true iff the player is currently in a battle
+     * @return true iff the player is currently in a battle
+     */
+
+    public boolean hasCurrEncounter(){
```

Comment:
Is it necessary to make a new method for this when you can just call the getter method and check if the returned value is null?

Reply:
Actually, maybe it is necessary to remove possible dependencies on Battle.